python/src/aica_api/client.py

```diff
@@ -114,13 +114,18 @@ def check(self) -> bool:
 
         version_info = semver.parse_version_info(self._api_version)
 
-        if version_info.major == 3:
+        if version_info.major == 4:
             return True
-        elif version_info.major > 3:
+        elif version_info.major > 4:
             self._logger.error(f'The detected API version v{self._api_version} is newer than the maximum API version '
```

Comment:
The `/version` endpoint now returns the CORE version, not the API version (which is determined simply by the `/protocol` endpoint as v2. So the compatibility checks and tables still make sense moving forward, but we should change the language of these messages:

"The detected core version ..."

Also, we should refactor `api_version()` to be called `core_version()` instead (and in that sense mark `api_version()` as deprecated), and finally add a `self.protocol()` method to return the result of `/protocol`.

Reply:
Should we also rename the decorator to `_requires_core_version` ?